### DISCUSSION TITLE
Network Video Recorder/server crashing when using 3/4 video feeds

### DISCUSSION THREAD

**Discussion by bradelvy** [bluenviron/mediamtx#1371] [Questions] [answered]
Hi, 

I have an rtsp simple server set up on a linux machine, with 4 webcams attached. 
They stream to the same port on the same server.
![image](https://user-images.githubusercontent.com/40993851/211581103-02980cff-1230-4dec-996b-63f5883606da.png)

These are then streamed to an NVR which pushes them onto a screen.
The problem is that the NVR crashes when there are too many attached.
I was thinking that the NVR cant handle 4 streams from the same ip and port, is there a way to have each camera stream on a different port, or run multiple servers for each camera?

Really appreciate any guidance!

Cheers,
Brad

**Comment by aler9** [comment, marked as answer]
Hello,

> I was thinking that the NVR cant handle 4 streams from the same ip and port

In the RTSP protocol, the port you see in the URL is used only during the handshake phase; a dedicated port is negotiated between rtsp-simple-server and the NVR in order to stream the video feeds.

If the NVR is not able to pick a proper port for each incoming feed (but anyway, are you sure that's the issue?), try using the TCP transport protocol:

```yml
protocols: [tcp]
```

Otherwise, you can allocate 4 different servers by using docker, on different ports:

```
docker run --rm -it -p 8554:8554 -e RTSP_PROTOCOLS=tcp -d aler9/rtsp-simple-server
docker run --rm -it -p 8555:8554 -e RTSP_PROTOCOLS=tcp -d aler9/rtsp-simple-server
docker run --rm -it -p 8556:8554 -e RTSP_PROTOCOLS=tcp -d aler9/rtsp-simple-server
docker run --rm -it -p 8557:8554 -e RTSP_PROTOCOLS=tcp -d aler9/rtsp-simple-server
```

**Reply by bradelvy** [reply]
Thank you! I will try this, its possible the server simply can't hack the data rate, but I think your above method is an improvement either way.

[a comment by an automated account is not shown]
